core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3322,49 +3363,69 @@ private SyncResult syncInodeMetadata(RpcContext rpcContext, LockedInodePath inod
         }
       }
       if (syncPlan.toLoadMetadata()) {
-        loadMetadata = true;
+        AlluxioURI mountUri = new AlluxioURI(mMountTable.getMountPoint(inodePath.getUri()));
+        // We only sync up to the mount point or the originalPath, whichever is lower in the tree
+        if (PathUtils.hasPrefix(originalPath.getUri().getPath(), mountUri.getPath())) {
+          pathsToLoad.add(originalPath.getUri().getPath());
+        } else {
+          pathsToLoad.add(mountUri.getPath());
+        }
       }
       if (syncPlan.toSyncChildren() && inode instanceof InodeDirectory
           && syncDescendantType != DescendantType.NONE) {
         UfsStatus[] listStatus = ufs.listStatus(ufsUri.toString());
```

Comment:
Can you move this down to right before we use it?

Reply:
Not sure what you mean..

Reply:
Could you clarify this? I am not sure what you mean.. what are we moving? down to where?